Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -366,6 +369,17 @@ protected void MuzzleFlash(EntityUid gun, AmmoComponent component, EntityUid? us
         CreateEffect(gun, ev, user);
     }
 
+    public void CauseImpulse(Vector2 direction, EntityUid? user = null)
+    {
+        if (user == null) return;
+        var userPhysics = EnsureComp<PhysicsComponent>(user.Value);
+        if (userPhysics == null) return;
+
+        if (!Gravity.IsWeightless(user.Value, userPhysics)) return;
+        var impulseVector =  direction.Normalized * userPhysics.Mass;
+        Physics.ApplyLinearImpulse(userPhysics, -impulseVector * 1.2f);
+    }
+
```

Comment:
1. Don't use inline returns
2. The EnsureComp should be a resolve as it shouldn't be adding it to a user.
3. This isn't getting called in shared so it still mispredicts.

Reply:
is this alright?